src/dense.jl

```diff
@@ -1,193 +1,207 @@
 # A higher level stepper, defined as a wrapper around another stepper.
 
-#TODO: how about having an DenseStepper <: AbstractWrapper <: AbstractStepper?
-immutable DenseStepper <: AbstractStepper
-    solver::Solver
+"""
+
+Dense output options:
+
+- tspan    ::Vector{T}  output times
+- points   ::Symbol which points are returned: `:specified` only the
+  ones in tspan or `:all` which includes also the step-points of the solver.
+- stopevent  Stop integration at a zero of this function
+- roottol    TODO
+
+"""
+
+immutable DenseOptions{T<:Number} <: Options{T}
+    tspan    ::Vector{T}
+    tstop    ::T
+    # points   ::Symbol
+    # stopevent::S
+    # roottol  ::T
 end
 
-solve(ode::ExplicitODE,
-      stepper::DenseStepper,
-      options::Options) = Solver(ode,stepper,options)
+@compat function (::Type{DenseOptions{T}}){T}(;
+                                              tstop         = T(Inf),
+                                              tspan::Vector = T[tstop],
+                                              # points::Symbol= :all,
+                                              # stopevent::S  = (t,y)->false,
+                                              # roottol       = eps(T)^T(1//3),
+                                              kargs...)
+    DenseOptions{T}(tspan,tstop)
+end
 
-dense(sol::Solver) = solve(sol.ode, DenseStepper(sol), sol.options)
 
 """
 
-The state of the dense stepper
+A stepper specialized in dense output.  It wraps around another
+`Solver` and stores the subsequent steps generated by `Solver` and
+interpolates the results on request (currently this means at the
+output times stored in `options.tspan`).
+
+"""
+immutable DenseStepper{S<:Solver,O<:DenseOptions} <: AbstractStepper
```

Comment:
Usually steppers are integrators, but this here is something else.  Confusing.  Does it need to be an `AbstractStepper` for any functionality?  Answer: no.